deps/thpool/thpool.c

```diff
@@ -121,50 +147,120 @@ static void bsem_post(struct bsem* bsem_p);
 static void bsem_post_all(struct bsem* bsem_p);
 static void bsem_wait(struct bsem* bsem_p);
 
-/* ========================== THREADPOOL ============================ */
+static bool pause_all() {
+  return g_pause_all;
+}
+static bool resume_all() {
+  return g_resume_all;
+}
+/* ========================== GENERAL ============================ */
+/* Register the process to a signal handler */
+void register_process_to_pause_handler(LogFunc log_cb) {
```

Comment:
I think this should be located somewhere on RediSearch code no? Not on thpool.c?